src/clientSideScene/segments.ts

```diff
@@ -144,6 +156,55 @@ export interface SegmentUtils {
   ) => CreateSegmentResult['updateOverlaysCallback'] | Error
 }
 
+/**
+ * Checks if a segment is fully constrained by examining all its constraint info
+ */
+function isSegmentFullyConstrained(
+  pathToNode: PathToNode,
+  ast: Program,
+  code: string
+): boolean {
+  try {
+    const nodeMeta = getNodeFromPath<any>(ast, pathToNode)
+    if (err(nodeMeta) || nodeMeta.node.type !== 'CallExpressionKw') {
+      return false
+    }
+
+    const constraintInfos = getConstraintInfoKw(nodeMeta.node, code, pathToNode)
```

Comment:
I think this is specific to this branch, I'm getting a ton of `Could not tell if this call to close was absolute Error: Unknown sketch function close` errors when playing in sketch mode, and I think they correspond to moments when the sketch constraint coloring isn't getting applied properly?